src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -1825,6 +1826,12 @@ because xbuild doesn't support framework reference assemblies.
 	</_CompileDexDependsOn>
 </PropertyGroup>
 
+<Target Name="_SetProguardMappingFileProperty">
+	<PropertyGroup>
+		<AndroidProguardMappingFile Condition=" '$(AndroidLinkTool)' == 'r8' And '$(AndroidProguardMappingFile)' != 'None' ">$(OutputPath)mapping.txt</AndroidProguardMappingFile>
```

Comment:
I'm not sure these semantics quite make sense?

If someone builds with:

```
msbuild /t:SignAndroidPackage /p:AndroidProguardMappingFile=my-custom-mapping.txt
```

what is supposed to happen?  This isn't a "documented" scenario, but with a property name ending in "file" that doesn't imply a "boolean" value, this looks quasi-reasonable?  Except values on the command-line aren't actually modifiable, meaning `$(AndroidProguardMappingFile)` would remain `my-custom-mapping.txt`, and I'm not sure what that would *do*.

I think a "better" idea would be to use a "boolean" value, e.g. `$(AndroidCreateProguardMappingFile)` (=True or False, default to True), and use a "private" `$(_AndroidProguardMappingFile)` to contain the filename:

```xml
<AndroidCreateProguardMappingFile Condition=" '$(AndroidCreateProguardMappingFile)' == '' And '$(AndroidLinkTool)' == 'r8' ">True</AndroidCreateProguardMappingFile>
<_AndroidProguardMappingFile Condition=" '$(AndroidCreateProguardMappingFile)' == 'True' ">$(OutputPath)mapping.txt<_AndroidProguardMappingFile>
```